main.cpp

```diff
@@ -401,90 +445,189 @@ struct Engines
+        int thrustLevel = 0;
 
-        void thrust(std::string whichTurbine);
-        void efficiency(std::string targetSpeed);
-        void noiseLevel(std::string whichTurbine);
+        void increaseThrustLevel(int targetLevel);
+        void increaseSpeed(int targetSpeed);
+        void increasePressure(int targetPressure);
     };
 
+    bool engageThrust(Turbines thrust,bool turbinesEngaged);
+    void increasePower(float amountOfIncrease);
+    void controlSpeed();
+};
+
+void Engines::Turbines::increaseThrustLevel(int targetLevel)
+{
+    if(targetLevel == 20)
+    {
+        thrustLevel += 20;
+    }
+    else
+    {
+    }
+}
+void Engines::Turbines::increaseSpeed(int targetSpeed)
+{
+    if(targetSpeed == 20)
+    {
+        currentSpeed += 20;
+    }
+}
+void Engines::Turbines::increasePressure(int targetPressure)
+{
+    if(targetPressure == 40)
+    {
+        currentPressure += 40;
+    }
+}
 
-//     3 things it can do:
-//         1) Provide thrust
-    void engageThrust(Turbines turbines);
-//         2) Generate elecricity
-    void increasePower(Turbines turbines, int amountOfIncrease);
-//         3) Control speed and altitude
-    void controlSpeedAndAltitude(Turbines turbines, int amountOfChange);
+bool Engines::engageThrust(Turbines thrust, bool turbinesEngaged)
+{
+    if(turbinesEngaged == true)
+    {
+        int targetLevel = 20;
+        thrust.increaseThrustLevel(targetLevel);
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+void Engines::increasePower(float amountOfIncrease)
+{
+    if(amountOfIncrease == 40.f)
+    {
+        powerOutput += 40.f;
+    }
+}
+void Engines::controlSpeed()
+{
+    if(engineSpeed == 0)
+    {
+        engineSpeed += 10;
+    }
+}
 
-    Turbines engaged;
-};
 
+// UDT 7
 struct CargoSpace
 {
-    // 5 properties:
-    //     1) Volume (float)
     float volume = 64.f;
-    //     2) Weight (int)
     int weight = 1300;
-    //     3) Temperature (int)
     int temperature = 75;
-    //     4) Capacity (int)
     int capacity = 400;
-    //     5) Loading time (float)
     float loadingTime = 45;
-    // 3 things it can do:
-    //     1) Transport
+
     void transport();
-    //     2) Store
-    void store();
-    //     3) Secure
-    void secure();
+    void store(int amountOfCargo);
+    void secure(bool isDoorEnclosed);
 };
 
+void CargoSpace::transport()
+{
+    std::cout << "Cargo is being transported." << std::endl;
+}
+
+void CargoSpace::store(int amountOfCargo)
+{
+    if(amountOfCargo <= capacity)
+    {
+    std::cout << amountOfCargo << " units of cargo stored." << std::endl;
+    }
+    else
+    {
+    std::cout << "Cargo exceeds maximum capacity." << std::endl;
+    }
+}
+
+void CargoSpace::secure(bool isDoorEnclosed)
+{
+    if(isDoorEnclosed)
+    {
+    std::cout << "Cargo is secured." << std::endl;
+    }
+    else
+    {
+    std::cout << "Cargo is not secured. Please close the door." << std::endl;
+    }
+}
```

Comment:
Code inside of Curly braces needs to be indented for spaces relative to those Curly braces
- [ ] revise